webviews/wkwebview/wkwebview/LinkViewController.swift

```diff
@@ -2,138 +2,185 @@
-        } else if (navigationAction.navigationType == WKNavigationType.linkActivated &&
-            (actionScheme == "http" || actionScheme == "https")) {
+            return
+        }
+
+        // Check if the url is our known redirect url. Universal links are not triggered when the WKWebView is running
+        // in the universal link's target application. Manually handle this case now.
+        if handleRedirectURL(url) {
+            decisionHandler(.cancel)
+            return
+        }
+
+        if navigationAction.navigationType == WKNavigationType.linkActivated &&
+            (actionScheme == "http" || actionScheme == "https") {
             // Handle http:// and https:// links inside of Plaid Link,
             // and open them in a new Safari page. This is necessary for links
             // such as "forgot-password" and "locked-account"
             UIApplication.shared.open(navigationAction.request.url!, options: convertToUIApplicationOpenExternalURLOptionsKeyDictionary([:]), completionHandler: nil)
             decisionHandler(.cancel)
-        } else {
-            print("Unrecognized URL scheme detected that is neither HTTP, HTTPS, or related to Plaid Link: \(navigationAction.request.url?.absoluteString)");
-            decisionHandler(.allow)
+            return
         }
+
+        // Allow anything not explicitly handled above.
+        decisionHandler(.allow)
+    }
+
+    // MARK: Private
+
+    private let webView = WKWebView()
+
+    /// generateLinkInitializationURL :: create the link.html url with query parameters
+    func generateLinkInitializationURL() -> URL {
+        // Create a new link_token via the /link/token/create endpoint. You will be
+        // able to configure this link_token to control Link behavior. To learn more
+        // about how to create a link_token, visit https://plaid.com/docs/#create-link-token.
+        //
+        // After creating a link_token, replace <#GENERATED_LINK_TOKEN#> with it below.
+        let config = [
+            "token": "<#GENERATED_LINK_TOKEN#>",
+            "isMobile": "true",
+            "isWebview": "true",
+        ]
+
+        // Build a dictionary with the Link configuration options
+        // See the Link docs (https://plaid.com/docs/quickstart) for full documentation.
+        var components = URLComponents()
+        components.scheme = "https"
+        components.host = "cdn.plaid.com"
+        components.path = "/link/v2/stable/link.html"
+        components.queryItems = config.map { URLQueryItem(name: $0, value: $1) }
+        return components.url!
+    }
+
+    /// getUrlParams - parse query parameters into a Dictionary
+    private func getUrlParams(url: URL) -> Dictionary<String, String> {
+        var paramsDictionary = [String: String]()
+        let queryItems = URLComponents(string: (url.absoluteString))?.queryItems
+        queryItems?.forEach { paramsDictionary[$0.name] = $0.value }
+        return paramsDictionary
```

Comment:
🖼️ This could possibly be more swifty:
```suggestion
        let queryItems = URLComponents(string: (url.absoluteString))?.queryItems
        return Dictionary(
            uniqueKeysWithValues: zip(
                queryItems.map { return $0.name },
                queryItems.map { return $0.value }
            )
        )
```